submissions/NJul/js-oop/app.js

```diff
@@ -0,0 +1,143 @@
+const Inhabitant = function (x, y, speed) {
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+};
+
+// Renders the enemy and the player into the game
+Inhabitant.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Inhabitant.prototype.update = function (dt) {};
+
+// Enemies our player must avoid
+const Enemy = function (x, y, speed) {
+  Inhabitant.call(this, x, y, speed);
+  // The following variables are used to determine the x and y axis and speed of the enemy
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+
+  // The image of the enemy of cockroach that is added to the playing field
+  this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype = Object.create(Inhabitant.prototype);
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+  // Multiplies the speed by the dt parameter on the x axis
+  this.x += this.speed * dt;
+
+  // Once enemies are off the canvas, they reappear randomly with different speeds
+  if (this.x > 510) {
+    this.x = -50;
+    this.speed = 100 + Math.floor(Math.random() * 400);
+  }
+
+  // Checks for collisions between the player and the enemies
+  if (
+    player.x < this.x + 80 &&
+    player.x + 80 > this.x &&
+    player.y < this.y + 60 &&
+    60 + player.y > this.y
+  ) {
+    player.x = 202;
+    player.y = 405;
+    score.updateMisses();
+  }
```

Comment:
`update` method does two different things here:
1. Update the position of the character (just like method name promises)
2. Check if the character collides with someone else (somethig the method name doesn't promise)

This is a clear sign that you need to split this into two methods.
Separation of Concerns principle should be followed